src/lib.rs

```diff
@@ -0,0 +1,111 @@
+extern crate tokio_trace;
+
+struct Span {
+    label: String,
+    ref_count: usize,
+}
+
+use std::fmt;
+use std::sync::Mutex;
+
+use tokio_trace::field::{Field, Visit};
+use tokio_trace::metadata::Level;
+use tokio_trace::{span, Event, Metadata, Subscriber};
+pub struct Visitor;
+
+impl Visit for Visitor {
+    fn record_debug(&mut self, field: &Field, value: &fmt::Debug) {
+        if field.name() == "message" {
+            print!("{:?}: ", value);
+        } else {
+            print!("{} = {:?} ", field.name(), value);
+        }
+    }
+}
+
+pub struct DemoSubscriber {
+    spans: Mutex<Vec<Span>>,
```

Comment:
You may not be particularly concerned about performance since this is just a demo, but it's worth noting that both the spans and the current span context will likely be read from more often than they're written to (i.e. you read from them in `Subscriber::event` but only write to them in `Subscriber::{new_span, enter, exit}`). So you will likely get better performance using a read-write lock rather than a mutex.

Reply:
Also, for the `tokio-console` project in particular, it shouldn't actually be necessary to store data in-process for each span. Instead, we can just send messages to the external aggregator with span IDs, and use those IDs to hydrate any dynamic data associated with that span.

This won't really work for a demo subscriber that's just printing to stdout (if you want to print dynamic data every time something happens in a span), but it's worth mentioning now.

Reply:
Yes, a read-write lock would make more sense. But when performance is a concern, we should look into `parking_lot` at first and not std. Whilste their primitives are being [integrated into std](https://github.com/rust-lang/rust/pull/56410), most rust versions `>1.26.0` will still lack in performance.

Regarding out-of-process data for spans, i haven't thought about that yet, that would make a pretty lightweight subscriber!

Reply:
> Yes, a read-write lock would make more sense. But when performance is a concern, we should look into `parking_lot` at first and not std. Whilste their primitives are being [integrated into std](https://github.com/rust-lang/rust/pull/56410), most rust versions `>1.26.0` will still lack in performance.

Yeah, that's correct --- I used `parking_lot`'s rwlocks in `tokio-trace-fmt`'s span store and saw a pretty significant performance improvement. As a side note, if you're interested in how a high-performance span store might look, you may want to take a look at [`fmt`'s implementation](https://github.com/tokio-rs/tokio-trace-nursery/blob/722edd3e10602bd064bf9c0d56ab6721e2069f55/tokio-trace-fmt/src/span.rs#L224-L501) --- I tried to make sure it was reasonably well-commented. 

But, for a demo, it's definitely not important to implement something like this, and since the span data used by `tokio-trace-console` will be stored out of process, it may not be something we ever end up worrying about!